pkg/controller/discovery/web/ns.go

```diff
@@ -117,29 +194,69 @@ func (h NsHandler) List(ctx *gin.Context) {
 //
 // Get a specific namespace on a cluster.
 func (h NsHandler) Get(ctx *gin.Context) {
-	status := h.Prepare(ctx)
-	if status != http.StatusOK {
-		ctx.Status(status)
-		return
+	ctx.Status(http.StatusMethodNotAllowed)
+}
+
+//
+// Has read access.
+func (h *NsHandler) hasRead(ns string) (bool, error) {
+	return h.rbac.Allow(&auth.Review{
+		Resource: auth.Namespace,
+		Verb:     auth.GET,
+		Name:     ns,
```

Comment:
it is weird but it looks like we need to have `Namespace` in the request... 

```
[alpatel@alpatel mig-operator]$ kubectl create -f - -o yaml << EOF
apiVersion: authorization.k8s.io/v1
kind: SubjectAccessReview
spec:
  user: bob
  resourceAttributes:
    group: ""
    resource: namespaces
    verb: "get"
    namespace: bob-destination
    name: bob-destination
  uid: 190531b5-590a-4ea6-98a4-9b45cdfeb094
  extra:
    scopes:
    - user:full
EOF
apiVersion: authorization.k8s.io/v1
kind: SubjectAccessReview
metadata:
  creationTimestamp: null
spec:
  extra:
    scopes:
    - user:full
  resourceAttributes:
    name: bob-destination
    namespace: bob-destination
    resource: namespaces
    verb: get
  uid: 190531b5-590a-4ea6-98a4-9b45cdfeb094
  user: bob
status:
  allowed: true
  reason: 'RBAC: allowed by RoleBinding "admin/bob-destination" of ClusterRole "admin"
    to User "bob"'
[alpatel@alpatel mig-operator]$ kubectl create -f - -o yaml << EOF
apiVersion: authorization.k8s.io/v1
kind: SubjectAccessReview
spec:
  user: bob
  resourceAttributes:
    group: ""
    resource: namespaces
    verb: "get"
    name: bob-destination
  uid: 190531b5-590a-4ea6-98a4-9b45cdfeb094
  extra:
    scopes:
    - user:full
EOF
apiVersion: authorization.k8s.io/v1
kind: SubjectAccessReview
metadata:
  creationTimestamp: null
spec:
  extra:
    scopes:
    - user:full
  resourceAttributes:
    name: bob-destination
    resource: namespaces
    verb: get
  uid: 190531b5-590a-4ea6-98a4-9b45cdfeb094
  user: bob
status:
  allowed: false
[alpatel@alpatel mig-operator]$
```

Reply:
Fixed.